third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -1132,8 +1132,18 @@ impl<'env> Generator<'env> {
                 if args.len() != pats.len() {
                     // Type checker should have complained already
                     self.internal_error(id, "inconsistent tuple arity")
+                } else if args.len() != 1 && self.have_overlapping_vars(pats, exp) {
+                    // Save each tuple arg (from rhs) into a temporary.
```

Comment:
Perhaps a quick sentence before why we do this, or perhaps just a simple example: "Prevent override in case of overlapping assignment, e.g. `(x, y) = (y, x)`"

Reply:
done